lib/driver.js

```diff
@@ -410,460 +391,399 @@ DatastaxDriver.prototype.executeCqlOnDriver = function executeCqlOnDriver(pool,
+    this._logQuery(cqlQuery, dataParams, options);
+    dataParams = this._normalizeParameters(dataParams);
 
+    if (isStream(callback)) {
+      // Keep passing the stream down the chain so we are still reusing logic
+      return this._execCqlStream(cqlQuery, dataParams, options, callback);
+    }
 
-var retryableResponseErrors = [
-  cqlDriver.types.responseErrorCodes.serverError,
-  cqlDriver.types.responseErrorCodes.unavailableException,
-  cqlDriver.types.responseErrorCodes.overloaded,
-  cqlDriver.types.responseErrorCodes.isBootstrapping,
-  cqlDriver.types.responseErrorCodes.writeTimeout,
-  cqlDriver.types.responseErrorCodes.readTimeout
-];
+    if (options.iterable) {
+      return this._execIterable(cqlQuery, dataParams, options);
+    }
 
-DatastaxDriver.prototype.canRetryError = function canRetryError(err) {
-  if (err instanceof cqlDriver.errors.ResponseError) {
-    return retryableResponseErrors.indexOf(err.code) !== -1;
+    const promise = this._execCql(cqlQuery, dataParams, options);
+    if (!callback) {
+      return promise;
+    }
+
+    return void promise.then(result => callback(null, result), callback);
   }
 
-  return (err instanceof cqlDriver.errors.NoHostAvailableError ||
-  err instanceof cqlDriver.errors.DriverInternalError);
-};
+  /*
+  namedQuery(queryName, dataParams, [options]: { consistency }, callback)
+  */
+  namedQuery(name, dataParams, options, callback) {
+    ({ options, callback } = checkOptionalParameters(options, callback));
 
-DatastaxDriver.prototype.query = DatastaxDriver.prototype.cql;
+    if (!this.queryCache) {
+      throw new Error('"queryDirectory" option must be set in order to use #namedQuery()');
+    }
 
-DatastaxDriver.prototype.select = function select(cql, dataParams, options, callback) {
-  var params = setDefaultConsistency(this.consistencyLevel.ONE, options, callback);
-  this.cql(cql, dataParams, params.options, params.callback);
-};
-DatastaxDriver.prototype.insert = function insert(cql, dataParams, options, callback) {
-  var params = setDefaultConsistency(this.consistencyLevel.LOCAL_QUORUM, options, callback);
-  this.cql(cql, dataParams, params.options, params.callback);
-};
-DatastaxDriver.prototype.update = function update(cql, dataParams, options, callback) {
-  var params = setDefaultConsistency(this.consistencyLevel.LOCAL_QUORUM, options, callback);
-  this.cql(cql, dataParams, params.options, params.callback);
-};
-DatastaxDriver.prototype.delete = function deleteQuery(cql, dataParams, options, callback) {
-  var params = setDefaultConsistency(this.consistencyLevel.LOCAL_QUORUM, options, callback);
-  this.cql(cql, dataParams, params.options, params.callback);
-};
+    const queryText = this.queryCache.readQuery(name);
+    options.queryName = options.queryName || name;
+    options.executeAsPrepared =
+      (this.config.supportsPreparedStatements && (typeof options.executeAsPrepared === 'undefined'))
```

Comment:
Remove `supportsPreparedStatements`. This is a holdover from `helenus`.

Reply:
@scommisso: done